### PR TITLE
Rename "kill" command to "delete"

### PR DESCRIPTION
**Pull request by bengeois**
Fix https://github.com/git-town/git-town/issues/4039

Continuing to work on issues related to rename command. Renamed all occurrences of the git town kill command to git town delete in the code, documentation and tests

Let me know if you'd like any adjustments!

### CODE REVIEW

File: website/src/commands/ship.md [hunk 1]
```diff
@@ -2,11 +2,11 @@
 
 > _git town ship [--to-parent] [--message &lt;text&gt;] [branch-name]_
 
-_Notice: Most people don't need to use the _ship_ command. The recommended way
+_Notice: Most people don't need to use the \_ship_ command. The recommended way
 to merge your feature branches is to use the web UI or merge queue of your code
 hosting service, as you would normally do. `git town ship` is for edge cases
 like developing in [offline mode](../preferences/offline.md) or when shipping
-[stacked changes](../stacked-changes.md)._
+[stacked changes](../stacked-changes.md).\_
```
[the remark this answers is not in the document]
Comment (bengeois) [reply website/src/commands/ship.md]: Mistake by automated IDE format, thanks 😀

**Comment by kevgo** [comment]
Thanks so much for implementing this! A significant polish of Git Town's user experience. Great work!

**Comment by kevgo** [comment]
Hey @bengeois, if you're up for it, you could take a shot at implementing @stephenwade’s suggestion in https://github.com/git-town/git-town/pull/4133#discussion_r1801610258. This would help avoid making it a breaking change, allowing us to get this release out sooner. No pressure at all though-—we’re already super thankful for all the work you’ve contributed so far!

To tackle suggestion 1, you could use the [undocumented deprecated new-pull-request command](https://github.com/git-town/git-town/blob/main/internal/cmd/new_pull_request.go) as a template. It can serve as a basis for creating a new Git Town command with the old name that calls the new command and prints a deprecation notice. This command is [hidden](https://github.com/git-town/git-town/blob/main/internal/cmd/new_pull_request.go#L24), so it doesn’t show up anywhere, and you won’t need to write end-to-end tests for it since it’ll be removed in one of the next breaking changes.

For suggestion 2, you just need to add the old and new keys to the [DeprecatedKeys map](https://github.com/git-town/git-town/blob/main/internal/config/configdomain/key.go#L156-L167) which will handle the upgrade automatically. To verify that the upgrade works, you can add an end-to-end test to the [Git configuration upgrade tests](https://github.com/git-town/git-town/tree/main/features/config/upgrade). See the [developer docs](https://github.com/git-town/git-town/blob/main/docs/DEVELOPMENT.md#end-to-end-tests) on how to run a single end-to-end test.

If you dive into any of these, feel free to tag me in a draft PR with your changes if you have any questions or need feedback. Cheers!

**Comment by bengeois** [comment]
Hi @kevgo, I’d be happy to take this on! Thanks for the detailed guidance and clarifications. I won’t be available at the end of this week, but I’ll take a look at it afterwards. 🙂

Cheers!

**Comment by kevgo** [comment]
Absolutely, thanks! I have created issues #4144, #4145, #4146, and #4147 to track these planned changes. This way, we can ensure we implement everything needed for the next [release](https://github.com/git-town/git-town/milestone/14). Let’s keep the momentum going!

**Pull request merged**